platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileManager.java

```diff
@@ -216,6 +216,74 @@ public int getTaskQueueSize() {
+     *
+     * @param reservedState the state that was written to disk
+     * @param directory     the directory where the state was written
+     * @param start         the nano start time of the state writing process
+     */
+    private void stateWrittenToDisk(
+            @NonNull final SignedState reservedState, @NonNull final Path directory, final long start) {
+
+        final long round = reservedState.getRound();
+        if (round > latestSavedStateRound.get()) {
+            latestSavedStateRound.set(round);
+        }
+
+        metrics.getWriteStateToDiskTimeMetric().update(TimeUnit.NANOSECONDS.toMillis(time.nanoTime() - start));
+        statusActionSubmitter.submitStatusAction(new StateWrittenToDiskAction(round));
+        stateToDiskAttemptConsumer.stateToDiskAttempt(reservedState, directory, true);
+    }
+
+    /**
+     * Checks whether a given state should be saved to disk
+     *
+     * @param reservedState        the state to potentially be saved
+     * @param outOfBand            whether this state save attempt is out-of-band
+     * @param stateLacksSignatures whether the state lacks signatures
+     * @return true if the state should be saved, false otherwise
+     */
+    private boolean shouldStateBeSaved(
+            final @NonNull SignedState reservedState, final boolean outOfBand, final boolean stateLacksSignatures) {
+
+        if (outOfBand) {
+            // states being saved out-of-band are exempt from the hasStateBeenSavedToDisk() check,
+            // and should always be saved
+            return true;
+        }
+
+        if (reservedState.hasStateBeenSavedToDisk()) {
+            logger.info(
+                    STATE_TO_DISK.getMarker(),
+                    "Not saving signed state for round {} to disk because it has already been saved.",
+                    reservedState.getRound());
+
+            return false;
+        } else {
+            if (stateLacksSignatures) {
+                metrics.getTotalUnsignedDiskStatesMetric().increment();
+                final long newCount = metrics.getTotalUnsignedDiskStatesMetric().get();
+
+                logger.error(
+                        EXCEPTION.getMarker(),
+                        new InsufficientSignaturesPayload(
+                                ("State written to disk for round %d did not have enough signatures. "
+                                                + "Collected signatures representing %d/%d weight. "
+                                                + "Total unsigned disk states so far: %d.")
+                                        .formatted(
+                                                reservedState.getRound(),
+                                                reservedState.getSigningWeight(),
+                                                reservedState.getAddressBook().getTotalWeight(),
+                                                newCount)));
```

Comment:
It seems odd to me to put this log here (at least, with the current method name). In tests, this has a very nontrivial side effect of failing the validator, but the name of the method suggests something that is side effect free.

Reply:
I'll try to massage this into a more intuitive form